java/server/src/org/openqa/grid/internal/Registry.java

```diff
@@ -176,6 +194,12 @@ public void removeIfPresent(RemoteProxy proxy) {
       log.warning(String.format(
           "Proxy '%s' was previously registered.  Cleaning up any stale test sessions.", proxy));
 
+      // TODO refactor all checks like this and registration
+      if(System.getProperty("com.sun.management.jmxremote") != null) {
```

Comment:
Second time you've used this check. Extract into a boolean method: `isJmxEnabled` or something similar